src/Api/Serializer/RestSerializer.php

```diff
@@ -231,20 +231,24 @@ function (array $matches) use ($varDefinitions) {
            $relative = $this->appendQuery($opts['query'], $relative);
         }
 
-        // If endpoint has path, remove leading '/' to preserve URI resolution.
         $path = $this->endpoint->getPath();
-        //accounts for removal of bucket from requestUri path
+
+        //Accounts for trailing '/' in path when custom endpoint
+        //is provided to endpointProviderV2
         if ($this->api->isModifiedModel()
             && $this->api->getServiceName() === 's3'
         ) {
-            $relative = $path . $relative;
+            substr($path, -1) === '/' && substr($relative, 0, 1) === '/'
```

Comment:
this just removes it from the relative if the path has a trailing slash and the relative has a trailing?  I feel like there's different logic that could accomplish the same thing; maybe an if statement would be clearer here?  Like in pseudo-code:
```
if (substr($path, -1) === '/' && substr($relative, 0, 1) === '/') {
$path = rtrim($path, '/');
}
$relative = $path . $relative;
```
Of course this won't work if the $path can't be changed if it's used later, but it looks like it's not.  

Also, maybe change `substr($relative, 0, 1)` to `$relative[0]`